addons/sale_stock/wizard/sale_order_cancel_views.xml

```diff
@@ -5,12 +5,12 @@
         <field name="model">sale.order.cancel</field>
         <field name="inherit_id" ref="sale.sale_order_cancel_view_form"/>
         <field name="arch" type="xml">
-            <field name="display_invoice_alert" position="after">
+            <span id="display_invoice_alert" position="after">
                 <field name="display_delivery_alert" invisible="1"/>
-                <div attrs="{'invisible': [('display_delivery_alert', '=', False)]}">
+                <span attrs="{'invisible': [('display_delivery_alert', '=', False)]}">
                     Some products have already been delivered. Returns can be created from the Delivery Orders.
-                </div>
-            </field>
+                </span>
+            </span>
```

Comment:
Why are those changes needed?

Reply:
To get all 3 messages (in the worst-case scenario) in a single block, instead of 3 different div, it's visually better.